thunder/tests/test_transforms.py

```diff
@@ -530,6 +530,33 @@ def fn():
     assert any(("CUDAGraph" in bsym.sym.name) for bsym in thunder.last_traces(jfn)[-1].bound_symbols)
 
 
+@requiresCUDA
+def test_cudagraph_fw_bw():
+    import torch
+    import thunder
+    import litgpt
+    from torch.testing import make_tensor
+    from functools import partial
+    from thunder.transforms.cudagraph import CUDAGraphTransform
+
+    device = torch.device("cuda")
+
+    cfg = litgpt.Config.from_name("open_llama_3b", n_layer=2)
+    with device:
+        make = partial(make_tensor, low=0, high=255, device=device, dtype=torch.long, requires_grad=False)
+        shape = (1,) + (cfg.block_size,)
+
+        x = make(shape)
+        m = litgpt.GPT(cfg)
+
+    m = thunder.jit(m, transforms=[CUDAGraphTransform()])
+
+    o = m(x)
+    o.sum().backward()
+
+    assert torch.cuda.max_memory_allocated() / 1e9 < 8
```

Comment:
The change seems good, but maybe we can
- enable shared memory pool for the transform, to enable
- grabbing a memory snapshot and checking the reservations for the pool.
To my mind, that should be much more reliably a thing of this test than the max_memory_allocated.